aspnetcore/blazor/fundamentals/routing.md

```diff
@@ -532,8 +576,12 @@ Blazor Web Apps are capable of two types of routing for page navigation and form
 * The feature isn't [explicitly disabled](xref:blazor/fundamentals/startup#disable-enhanced-navigation-and-form-handling).
 * The destination URL is within the internal base URI space (the app's base path).
 
+If server-side routing and enhanced navigation are enabled, [location changing handlers](#location-changes) are only invoked for programmatic navigations initiated from an interactive runtime. In future releases, additional types of navigations, such as link clicks, may also invoke location changing handlers.
+
 If enhanced navigation is available for interactive client routing, navigation always goes through the interactive client-side router. This point is only relevant in an uncommon scenario where an interactive `<Router>` is nested inside a server-side rendered `<Router>`. In that case, the interactive router takes priority when handling navigation, so enhanced navigation isn't used for navigation because it's a server-side routing feature.
 
+When an enhanced navigation occurs, [`LocationChanged` event handlers](#location-changes) registered with interactive server and WebAssembly runtimes are typically invoked. There are cases when location changing handlers might not intercept an enhanced navigation. For example, the user might switch to another page before an interactive runtime becomes available. Therefore, it's important that app logic not rely on invoking a location changing handler, as there's no guarantee of the handler executing. If the app must rely on a location change handler firing, don't use enhanced navigation, which is explained later in this section.
```

Comment:
This looks good. The thing that might need further clarification is:
> If the app must rely on a location change handler firing, don't use enhanced navigation, which is explained later in this section.

This makes it sound like the solution is to disable enhanced navigation, but actually won't make a difference. If they need location changing handlers to be reliably invoked for internal navigations, then they will need to use an interactive (client-side) router instead of server-side routing.

Reply:
I'm *still* becoming acquainted with these concepts, so for ...

> use an interactive client-side router instead of server-side routing

... how is that done? Must they make the entire component adopt interactive WebAssembly rendering?